python/cudf/cudf/tests/test_pandas_comparison.py

```diff
@@ -0,0 +1,366 @@
+        (5, "cats", "jump", np.nan),
+        (2, "dogs", "dig", 7.5),
+        (3, "cows", "moo", -2.1, "occasionally"),
+    ]
+    return lib.DataFrame(data)
+
+
+@pytest.fixture
+def dt():
+    return [
+        (5, "cats", "jump", np.nan),
+        (2, "dogs", "dig", 7.5),
+        (3, "cows", "moo", -2.1, "occasionally"),
+    ]
+
+
+@pandas_comparison_test
+def test_init4(lib, dt):
+    return lib.DataFrame(dt)
+
+
+@pandas_comparison_test
+def test_init7(dt, lib):
+    return lib.DataFrame(dt)
+
+
+@pandas_comparison_test
+@pytest.mark.parametrize(
+    "data",
+    [
+        [
+            (5, "cats", "jump", np.nan),
+            (2, "dogs", "dig", 7.5),
+            (3, "cows", "moo", -2.1, "occasionally"),
+        ]
+    ],
+)
+def test_init5(lib, data):
+    return lib.DataFrame(data)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        [
+            (5, "cats", "jump", np.nan),
+            (2, "dogs", "dig", 7.5),
+            (3, "cows", "moo", -2.1, "occasionally"),
+        ]
+    ],
+)
+@pandas_comparison_test
+def test_init6(lib, data):
+    return lib.DataFrame(data)
+
+
+@pytest.mark.parametrize(
+    "df",
+    [
+        pd.DataFrame(
+            [
+                (5, "cats", "jump", np.nan),
+                (2, "dogs", "dig", 7.5),
+                (3, "cows", "moo", -2.1, "occasionally"),
+            ]
+        )
+    ],
+)
+@pandas_comparison_test
+def test_from_pandas(lib, df):
+    return lib.from_pandas(df) if _is_cudf(lib) else df
+
+
+def custom_assert(expected, got):
+    cudf.testing.assert_frame_equal(cudf.from_pandas(expected), got)
+
+
+@pytest.mark.parametrize(
+    "df",
+    [
+        pd.DataFrame(
+            [
+                (5, "cats", "jump", np.nan),
+                (2, "dogs", "dig", 7.5),
+                (3, "cows", "moo", -2.1, "occasionally"),
+            ]
+        )
+    ],
+)
+@pandas_comparison_test(assert_func=custom_assert)
+def test_from_pandas_custom_assert(lib, df):
+    return lib.from_pandas(df) if _is_cudf(lib) else df
+
+
+@pytest.fixture
+def df():
+    return cudf.DataFrame(
+        [
+            (5, "cats", "jump", np.nan),
+            (2, "dogs", "dig", 7.5),
+            (3, "cows", "moo", -2.1, "occasionally"),
+        ]
+    )
+
+
+@pandas_comparison_test(cudf_objects="df")
+def test_fixture_concat(lib, df):
+    return lib.concat([df, df])
+
+
+@pytest.mark.parametrize(
+    "df",
+    [
+        cudf.DataFrame(
+            [
+                (5, "cats", "jump", np.nan),
+                (2, "dogs", "dig", 7.5),
+                (3, "cows", "moo", -2.1, "occasionally"),
+            ]
+        )
+    ],
+)
+@pandas_comparison_test(cudf_objects="df")
+def test_param_concat(lib, df):
+    return lib.concat([df, df])
+
+
+# TODO: Instead of requiring the user to pass `cudf_object`, we could require
+# the cudf object (fixture or parameter) to be the first argument. The current
+# approach seems more explicit, but perhaps we would prefer the other, less
```

Comment:
The explicit-ness is more like how pytest fixtures work too, afaik. I don't mind it.